### PR TITLE
PRESIDECMS-3323 inherited rule handler component metadata is ignored

### PR DESCRIPTION
**Pull request by choontat-pixl8**
Get the extended component metadata, including
- component properties (e.g. `@expressionContexts`)
- functions, function parameters (e.g. `required string type`) and their properties (e.g. `@type.fieldType`)

Also updates the expression test service to pass `Duplicate( meta.functions[1] )` to match the new changes

### CODE REVIEW

File: system/services/rulesEngine/RulesEngineExpressionReaderService.cfc [hunk 1]
```diff
@@ -114,37 +114,27 @@ component displayName="RulesEngine Expression Reader Service" {
...
-					, filterHandler         = filterObjects.len() ? "rules.expressions.#baseId#.prepareFilters" : ""
-					, labelHandler          = "rules.expressions.#baseId#.getLabel"
-					, textHandler           = "rules.expressions.#baseId#.getText"
-					, expressionHandlerArgs = {}
-					, filterHandlerArgs     = {}
-					, labelHandlerArgs      = {}
-					, textHandlerArgs       = {}
-				};
-
-			} else if ( func.name == "prepareFilters" ) {
```
Comment (Pixl8RusselCole) [inline system/services/rulesEngine/RulesEngineExpressionReaderService.cfc:140]: Was it your intention to remove prepareFilters completely?

File: tests/.gitignore [hunk 2]
```diff
@@ -3,3 +3,4 @@ web.config
 testbox
 testresults*.html
 .cfconfig.json
+server-presidetests.json
```
Comment (Pixl8RusselCole) [inline tests/.gitignore]: Not a biggy but I don't think this should be in this ticket.

File: system/services/rulesEngine/RulesEngineExpressionReaderService.cfc (hunk 1 above)
Comment (choontat-pixl8) [reply system/services/rulesEngine/RulesEngineExpressionReaderService.cfc:140]: The original `prepareFilters` check is to ensure that the `filterObjects` function property is set in the expression after the `evaluateExpression` function is processed.

The current `_getExtendedFunctionsMeta` function returns a struct of function meta, for example
```
{
      prepareFilters     = { objects="user,contact", parameters=[] }
    , evaluateExpression = { parameters=[] }
}
```

So we can now do 
```
var filterObjects = ListToArray( functions.prepareFilters.objects ?: "" );
...
expressions[ baseId ] = {
      filterObjects = filterObjects
    , filterHandler = filterObjects.len() ? "rules.expressions.#baseId#.prepareFilters" : ""
    ...
};
```

File: tests/.gitignore (hunk 2 above)
Comment (choontat-pixl8) [reply tests/.gitignore]: This line is now removed alongside with the `server-presidetests.json` file
